coordinator/persistence/aggregation/src/main/kotlin/net/consensys/zkevm/persistence/dao/aggregation/PostgresAggregationsDao.kt

```diff
@@ -296,6 +294,27 @@ class PostgresAggregationsDao(
       }
   }
 
+  override fun findHighestConsecutiveEndBlockNumber(
+    fromBlockNumber: Long
+  ): SafeFuture<Long?> {
+    return selectAggregations
+      .execute(
+        Tuple.of(
+          fromBlockNumber,
+          aggregationStatusToDbValue(Aggregation.Status.Proven),
+          Int.MAX_VALUE
+        )
+      )
+      .toSafeFuture()
+      .thenApply { rowSet ->
+        if (rowSet.size() > 0) {
+          rowSet.last().getLong("end_block_number")
```

Comment:
Using `lastOrNull()` we can remove this if and simplify this significantly

Reply:
Also revised that in other usages.